validate/validate_test.go

```diff
@@ -0,0 +1,82 @@
+package validate
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func checkErrors(t *testing.T, title string, msgs []string, valid bool) {
+	if valid && len(msgs) > 0 {
+		t.Fatalf("%s: expected not to get error, but get %d errors:\n%s", title, len(msgs), strings.Join(msgs, "\n"))
+	} else if !valid && len(msgs) == 0 {
+		t.Fatalf("%s: expected to get error, but actually not", title)
+	}
+}
+
+func TestCheckRootfsPath(t *testing.T) {
+	tmpBundle, err := ioutil.TempDir("", "oci-check-rootfspath")
+	if err != nil {
+		t.Fatalf("Failed to create a TempDir in 'CheckRootfsPath'")
+	}
+	defer os.RemoveAll(tmpBundle)
+
+	rootfsDir := "rootfs"
+	rootfsNonDir := "rootfsfile"
+	rootfsNonExists := "rootfsnil"
+	os.MkdirAll(filepath.Join(tmpBundle, rootfsDir), 0750)
```

Comment:
I just used to set 0750 to all the created dir. I don't mind to change it to 0700. Update comes soon.